### PR TITLE
Skeleton segments

### PR DESCRIPTION
**Pull request by ceesem**
Two improvements.

1) Mesh masking uses a trimesh-native approach that significantly speeds up the mask-based reindexing.
2) Exposes/allows a few new features of the skeleton graph, such as snipping graphs at a point, parent and child node look-ups, and better nomenclature for sets of covering paths. These are useful for other skeleton computations.

This does have a small breaking change in the order of arguments to trimesh_vtk.linked_point_actor, which no longer needs indices.

### CODE REVIEW

**Comment by fcollman** [comment]
need tests for the new functionality

File: meshparty/skeleton.py [hunk 1]
```diff
@@ -180,17 +185,60 @@ def reroot(self, new_root):
...
+    def downstream_nodes(self, vinds):
+        if np.isscalar(vinds):
+            vinds = [vinds]
+            return_single = True
+        else:
+            return_single = False
+
+        dns = []
+        for vind in vinds:
+            g = self.cut_graph(vinds)
```
Comment (fcollman) [inline meshparty/skeleton.py]: is cut graph the right idea here? or is it just a directed graph with the edges flipped?

File: meshparty/trimesh_vtk.py [hunk 2]
```diff
@@ -482,14 +482,18 @@ def point_cloud_actor(xyz,
...
     
     actor = vtk.vtkActor()
-    #mapper.ScalarVisibilityOff()
     actor.SetMapper(mapper)
     return actor
 
 
-def linked_point_actor(vertices_a, inds_a,
-                       vertices_b, inds_b,
+def linked_point_actor(vertices_a, vertices_b,
```
Comment (fcollman) [inline meshparty/trimesh_vtk.py]: this is technically a breaking change, but I don't think this is used that much right now so lets keep it.

File: meshparty/trimesh_io.py [hunk 3]
```diff
@@ -861,25 +860,23 @@ def filter_unmasked_boolean(self, unmasked_boolean):
...
         if include_outside:
-            long_shape = unmasked_shape.ravel()
-            within_mask = self.node_mask[long_shape].reshape(unmasked_shape.shape)
-            filtered_shape_all = np.full(unmasked_shape.shape, np.nan)
-            if within_mask.ndim==1:
-                filtered_shape_all[within_mask==True] = filtered_shape.squeeze()
-            else:
-                filtered_shape_all[np.all(within_mask==True, axis=1)] = filtered_shape
-            filtered_shape = filtered_shape_all.astype(int).squeeze()
+            new_shape[~keep_rows] = np.nan
```
Comment (fcollman) [inline meshparty/trimesh_io.py]: we need a test here.

File: meshparty/trimesh_vtk.py (hunk 2 above)
Comment (ceesem) [reply meshparty/trimesh_vtk.py]: Agreed on both counts.

File: meshparty/trimesh_io.py (hunk 3 above)
Comment (ceesem) [reply meshparty/trimesh_io.py]: Better to omit it entirely, I think. It's easy to replicate the overall goal (to know which indices are not inside the mask) in other ways and I have never used it.

File: meshparty/skeleton.py (hunk 1 above)
Comment (ceesem) [reply meshparty/skeleton.py]: the cut graph is fine, but I replaced connected components with Dijkstra on the graph transpose. Much faster for large trees than doing the same thing in python.

**Comment by fcollman** [comment]
looks good to me!

**Pull request merged**